hw02/EnemyCollision.cs

```diff
@@ -52,5 +64,44 @@ void Shoot ()
     Use the WaitForSeconds function to tell it to switch weapons every 5 seconds. 
 
     Remember to call your Coroutine.
-    */ 
-}
\ No newline at end of file
+    */
+
+    public class Weapon : MonoBehaviour
+    {
+        public int arrow = 0;
+        public int sword = 1;
+        public int rocket = 2;
+        public int[] weapons = new int[3];
+        public bool isFrozen = false;
+
+        private void Start()
+        {
+            weapons[0] = arrow;
+            weapons[1] = sword;
+            weapons[2] = rocket;
+        }
+
+        private void Update()
+        {
+            if (isFrozen == false)
+            {
+                for (int i = 0; i < weapons.Length; i++)
+                {
+                    Debug.Log("current weapon: " + weapons[i]);
+                    StartCoroutine(SwapWeapons());
```

Comment:
Close! We can't put coroutine inside Update(). The Update() is run every frame, so it will not achieve the expected result.